File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
#include<unordered_set>
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_set<int> numsSet(nums.begin(), nums.end());
        int result = 0;
        for(int num: numsSet){
            if(numsSet.find(num - 1) == numsSet.end()){
                int length = 1;
                while(numsSet.find(num + length) != numsSet.end())length += 1;
                result = max(result, length);
            }
        }
        return result;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> numsSet = new HashSet<Integer>();
        int result = 0;
        for(int num: nums){
            numsSet.add(num);
        }
        for(int num: numsSet){
            if(!numsSet.contains(num - 1)){
                int length = 1;
                while(numsSet.contains(num + length)) length += 1;
                result = Math.max(length, result);
            }
        }
        return result;
    }
}